### PR TITLE
feat(reload): reload PDF button, palette entry, keep/discard-annotations setting (#115)

### LINKED ISSUE #115: Reload PDF: toolbar button, command palette entry, keep-vs-discard-annotations setting
User needs a way to re-fetch the currently open PDF source (local file path re-read, or remote link re-fetched e.g. Google Slides) without closing the document.

Requirements:
1. Toolbar button (near page nav) that reloads the underlying PDF and re-renders pages.
2. Command palette entry: 'Reload PDF'.
3. Setting ('Reload behavior'): 'Keep annotations' (default) vs 'Discard annotations'. When discarding, clear per-page objects and reset history.
4. When keeping: preserve annotations on pages whose page count matches; warn and remap/truncate if the new PDF has a different page count.
5. Tests: reload command behavior, setting toggle, page-count-change edge cases.

### PR DESCRIPTION
**Pull request by Adamkadaban**
Closes #115. Toolbar reload button + `pdf.reload` palette command. New `PdfSource` discriminated union and `fetchPdfMeta(source)` as the single load dispatch point — both first-open and reload share it, so Slides (#114) will slot in cleanly. Setting `Reload behavior` (keep/discard). On keep + page-count mismatch, truncates and emits a warning toast; on discard, clears objects and resets history.

### CODE REVIEW

File: src/lib/pdf/reload.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    const sourceIndex = page.pdfSourceIndex ?? i;
+    const newDims = meta.pages[sourceIndex];
+    if (!newDims) return page;
+    return { ...page, width: newDims.width, height: newDims.height };
+  });
+
+  const appended: Page[] = [];
+  for (let i = previous.pages.length; i < meta.pageCount; i += 1) {
+    appended.push(emptyPdfPage(i, meta.pages[i]));
+  }
```
[the remark this answers is not in the document]
Comment (Adamkadaban) [reply src/lib/pdf/reload.ts]: Agreed — fixed in ff08e33. `buildKeptDoc` now bins previous pages into pdf-by-`pdfSourceIndex` and blanks-by-anchor maps, rebuilds pdf slots against `meta.pageCount`, and re-inserts blanks after their anchor PDF page.

File: src/lib/pdf/reload.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  const { doc, truncated } = buildKeptDoc(previous, meta);
+  if (!truncated) return { doc, warning: null };
+  return {
+    doc,
+    warning: `PDF now has ${meta.pageCount} page${
+      meta.pageCount === 1 ? '' : 's'
+    }; truncated ${previous.pages.length - meta.pageCount} trailing page${
+      previous.pages.length - meta.pageCount === 1 ? '' : 's'
+    } from the document.`,
+  };
```
[the remark this answers is not in the document]
Comment (Adamkadaban) [reply src/lib/pdf/reload.ts]: Agreed — fixed in ff08e33. Warning now counts `droppedPdfPages` (previous `pdf`-type entries whose `pdfSourceIndex >= meta.pageCount`), not array-length delta. Test covers the blank-heavy case.

File: src/lib/pdf/loader.ts [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    setError(err instanceof Error ? err.message : String(err));
+    return null;
+  }
+}
+
+export async function loadPdfFromSource(source: PdfSource): Promise<PdfMeta | null> {
+  setSource(source);
+  const meta = await fetchMetaWithState(source);
+  if (!meta) return null;
+
```
[the remark this answers is not in the document]
Comment (Adamkadaban) [reply src/lib/pdf/loader.ts]: Agreed — fixed in ff08e33. `setSource(source)` now runs only after `fetchMetaWithState` resolves. On failure the previous `pdf.source` is preserved; new test asserts this.

File: src/lib/pdf/loader.ts [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+  source: PdfSource;
+  behavior?: ReloadBehavior;
+}): Promise<ReloadResult | null> {
+  const behavior = opts.behavior ?? settings.snapshot().reloadBehavior;
+  const meta = await fetchMetaWithState(opts.source);
+  if (!meta) return null;
+
+  const { doc, warning } = planReload(opts.currentDoc, meta, behavior);
+  documentStore.load(doc);
+
```
[the remark this answers is not in the document]
Comment (Adamkadaban) [reply src/lib/pdf/loader.ts:107]: Agreed — fixed in ff08e33. Added `documentStore.replace(doc)` that swaps state without `history.clear()`. Keep uses `replace`; discard uses `load`. Tests confirm history survives `replace` and still clears on `load`.

File: src/routes/+page.svelte [hunk 5]
```diff
@@ -494,7 +449,6 @@
 
   onDestroy(() => {
     stopBridge?.();
-    stopAutosave?.();
     stopHydration?.();
     stopPresenterBridge?.();
     stopSidebarBridge?.();
```
[the remark this answers is not in the document]
Comment (Adamkadaban) [reply src/routes/+page.svelte:454]: Agreed — fixed in ff08e33. Exported `stopAutosave()` from `$lib/pdf/loader`, called from `onDestroy` in `+page.svelte`. Idempotent-stop test included.
